src/App.tsx

```diff
@@ -1,16 +1,24 @@
+import { FC, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
+
+export const App: FC = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const onAdd = ((movie: Movie) => {
+    setMovies(currentMovies => [...currentMovies, movie]);
+  });
 
-export const App = () => {
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={onAdd} />
```

Comment:
```suggestion
        <NewMovie onAdd={onAdd} key={count} />
```
If you change `count` your component will be created from scratch with default values and no errors messages.